src/identities/masterpubkey.ts

```diff
@@ -0,0 +1,342 @@
+import { AddressInterface } from '../types';
+import Identity, {
+  IdentityInterface,
+  IdentityType,
+  IdentityOpts,
+} from '../identity';
+import { BufferMap, isValidXpub, isValidBlindPub, toXpub } from '../utils';
+import { BIP32Interface, fromBase58 } from 'bip32';
+import { Slip77Interface, fromMasterBlindingKey } from 'slip77';
+import { payments } from 'liquidjs-lib';
+
+export interface MasterPublicKeyOptsValue {
+  masterPublicKey: string;
+  masterBlindingKey: string;
+}
+
+function instanceOfMasterPublicKeyOptsValue(
+  value: any
+): value is MasterPublicKeyOptsValue {
+  return 'masterPublicKey' in value && 'masterBlindingKey' in value;
+}
+
+interface AddressInterfaceExtended {
+  address: AddressInterface;
+  publicKey: string;
+  derivationPath: string;
+}
+
+// util function that parse a derivation path and return the index
+function getIndex(addrExtended: AddressInterfaceExtended) {
+  const derivationPathSplitted = addrExtended.derivationPath.split('/');
+  const index: number = parseInt(
+    derivationPathSplitted[derivationPathSplitted.length - 1]
+  );
+
+  return index;
+}
+
+export class MasterPublicKey extends Identity implements IdentityInterface {
+  static INITIAL_BASE_PATH: string = "m/84'/0'/0'";
+  static INITIAL_INDEX: number = 0;
+
+  private baseDerivationPath: string = MasterPublicKey.INITIAL_BASE_PATH;
+  private index: number = MasterPublicKey.INITIAL_INDEX;
+  private changeIndex: number = MasterPublicKey.INITIAL_INDEX;
+  private scriptToAddressCache: BufferMap<
+    AddressInterfaceExtended
+  > = new BufferMap();
+
+  readonly masterPublicKeyNode: BIP32Interface;
+  readonly masterBlindingKeyNode: Slip77Interface;
+
+  readonly isRestored: Promise<boolean>;
+
+  constructor(args: IdentityOpts) {
+    super(args);
+
+    // check the identity type
+    if (args.type !== IdentityType.MasterPublicKey) {
+      throw new Error(
+        'The identity arguments have not the MasterPublicKey type.'
+      );
+    }
+    // check the arguments
+    if (!instanceOfMasterPublicKeyOptsValue(args.value)) {
+      throw new Error(
+        'The value of IdentityOpts is not valid for MasterPublicKey Identity.'
+      );
+    }
+    // validate xpub
+    if (!isValidXpub(args.value.masterPublicKey, this.network)) {
+      throw new Error('Master public key is not valid');
+    }
+    // validate master blinding key
+    if (!isValidBlindPub(args.value.masterBlindingKey)) {
```

Comment:
the master blinding key is an extended private key, hence the name of its variable is not appropriate:
```suggestion
    if (!isValidExtendedBlindKey(args.value.masterBlindingKey)) {
```

Reply:
cc @louisinger

Reply:
This is missing?

Reply:
done :)